### ISSUE TITLE
Incorrect GridMap cursor offset in 3D Half Resolution view

### ISSUE THREAD

**Issue by kkga** [godotengine/godot#33937]
**Godot version:**
3.2 beta 2 official

**OS/device including version:**
macOS Catalina 10.15.1

**Issue description:**
When "Half Resolution" option is enabled in the 3D view settings, the GridMap offsets the cursor by 2x from the top-left of viewport.

![2019-11-27 13 55 32](https://user-images.githubusercontent.com/1460122/69717935-1f4c3300-111e-11ea-947d-919e26821572.gif)

**Steps to reproduce:**
1. Add a GridMap
2. Enable "Half Resolution" in 3D view settings
3. Try to paint on the GridMap

**Comment by Calinou** [comment]
Related to #13364 (which is about 3D selection in general). Can you reproduce this when selecting 3D nodes in the viewport?

**Comment by rbarongr** [comment]
This is still happening in v4.0.2.stable.official [7a0977ce2], macOS Ventura 13.0.1.


https://user-images.githubusercontent.com/91340577/233439815-c685a748-c4dd-4a82-bf13-22acd7cc9758.mp4

**Comment by Souleymeine** [comment]
From https://github.com/godotengine/godot/issues/106973 ( @Calinou )
> Thanks for the report! Consolidating in https://github.com/godotengine/godot/issues/33937.
(I choose to reply here as my issue has been marked as archived)

Sorry for the late reply.
I'm a little confused regarding the state of this issue. Has a PR been opened to resolve the issue?
I tried to fix it myself using one of your older draft PR (https://github.com/godotengine/godot/pull/33940), "porting" it to version 4.5.dev  but the issue you talked about in this PR still remains: using multiple viewports breaks the fix, as only the shrink factor of the first viewport is used. I tried ways to find the current focused viewport, but without success, otherwise I would've created my own PR. What I don't understand is that this bug was first reported 6 years ago, and I don't know if it keeps coming back or if it has never been fixed. 
Have a great rest of your day!

**Comment by Calinou** [comment]
> I'm a little confused regarding the state of this issue. Has a PR been opened to resolve the issue?

Yes, https://github.com/godotengine/godot/pull/93436 fixes this issue (including when multiple viewports are used). It needs a review before it can be merged.

> What I don't understand is that this bug was first reported 6 years ago, and I don't know if it keeps coming back or if it has never been fixed.

The bug has always been around ever since Half Resolution was added in 3.0.
